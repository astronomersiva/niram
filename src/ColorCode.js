import React, { Component } from 'react';
import { hot } from 'react-hot-loader';
import { css } from 'react-emotion';

const hexCodeStyle = css`
  height: 12vh;
  font-size: 8vh;
  padding: 5vh;
`;

class ColorCode extends Component {
  rgbToHex = rgb => {
    const hexArray = rgb.map(component => {
      const base16 = component.toString(16);
      return base16.length === 1 ? `0${base16}` : base16;
    });

    return `#${hexArray.join('')}`;
  };

  render() {
    const { activeColor } = this.props;

    return (
      <div className={hexCodeStyle}>
        {activeColor.length ? this.rgbToHex(activeColor) : ''}
      </div>
    );
  }
}

export default hot(module)(ColorCode);
