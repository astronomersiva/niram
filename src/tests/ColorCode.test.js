import React from 'react';
import renderer from 'react-test-renderer';
import ColorCode from '../ColorCode';

describe('<ColorCode />', () => {
  it('converts rgb array to hex code', () => {
    const tree = renderer
      .create(<ColorCode activeColor={[255, 0, 0]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('is empty when no rgb array is passed', () => {
    const tree = renderer.create(<ColorCode activeColor={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
