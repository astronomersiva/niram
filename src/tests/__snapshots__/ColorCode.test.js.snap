// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorCode /> converts rgb array to hex code 1`] = `
<div
  className="css-185dxra"
>
  #ff0000
</div>
`;

exports[`<ColorCode /> is empty when no rgb array is passed 1`] = `
<div
  className="css-185dxra"
>
  
</div>
`;
